microSALT/utils/referencer/utils.py

```diff
@@ -0,0 +1,38 @@
+from microSALT.utils.referencer.constants import (
+    Escherichia,
+    Klebsiella,
+    ESCHERICHIA_REFERENCE,
+    KLEBSIELLA_REFERENCE,
+)
+
+
+def get_species(organism_name: str) -> str:
+    """Returns the species of the organism name."""
+    return organism_name.lower().split()[-1]
+
+
+def is_escherichia(species_name: str) -> bool:
+    """Checks if the organism name is a proper Escherichia."""
+    return any(species_name == member.value for member in Escherichia)
+
+
+def is_klebsiella(species_name: str) -> bool:
+    """Checks if the organism name is a proper Klebsiella."""
+    return any(species_name == member.value for member in Klebsiella)
+
+
+def get_reference_if_enterobacteriaceae(organism_name: str) -> str:
+    """Returns the reference for the organism name if it belongs to Enterobacteriaceae."""
+    species: str = get_species(organism_name)
+    GENUS_REFERENCE_MAP = {
+        "escherichia": (is_escherichia, ESCHERICHIA_REFERENCE),
+        "klebsiella": (is_klebsiella, KLEBSIELLA_REFERENCE),
+    }
+    return next(
```

Comment:
This code is very unclear to me, and would benefit from a clarifying comment.

Does the next() call here mean that the first item of the tuple being used? Will this tuple always only contain one item?

Don't you think it would be clearer with just:

```python
references = (
    reference
    for check_func, references in GENUS_REFERENCE_LIST
    if check_func(species_name=species)
)
if refereces:
    return references
return organism_name
```